net/net.go

```diff
@@ -336,7 +337,9 @@ func (t *net) pullThreadUnsafe(ctx context.Context, id thread.ID) error {
 				log.Error(err)
 				return
 			}
+			lock.Lock()
```

Comment:
This is a very important one. Logs are pulled in parallel and they were appended to a shared slice that wasn't guarded. That can lead to unexpected behavior (loose pulled things, corruption, and wild things).

Reply:
Hrm, good catch. It doesn't look like the interior maps will collide, but the appending to the slice will definitely cause issues. Looks like it's from your original concurrency fixes: https://github.com/textileio/go-threads/commit/9deb080fa44326df16a695954f43d96c99c748c4

Reply:
Possibly, which solved a ton of other ones.

Reply:
Indeed! The whole thing was hot off the press :) Anyway, not pointing fingers... you just mentioned you were curious about the original intention of the logic.